Softeq.XToolkit.PushNotifications.Droid/Abstract/IPushNotificationsConsumer.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System.Threading.Tasks;
+using Android.Content;
+using Android.OS;
+using Firebase.Messaging;
+
+namespace Softeq.XToolkit.PushNotifications.Droid.Abstract
+{
+    /// <summary>
+    ///     Basic interface for push notification consumer on Android platform.
+    /// </summary>
+    public interface IPushNotificationsConsumer
+    {
+        /// <summary>
+        ///    Callback called when push notification is received.
+        /// </summary>
+        /// <param name="message">Received remote message.</param>
+        /// <returns><see langword="true"/> if consumer has handled remote message, <see langword="false"/> otherwise.</returns>
+        bool TryHandleNotification(RemoteMessage message);
```

Comment:
Not sure. IPushNotificationConsumer interface works not only with notifications, but also with tokens and registration callbacks - so it might not be obvious what exactly is being handled here (unless you know what is `RemoteMessage` class and what does it represent)